### PR TITLE
chore(upgrade): Kea 3.0, keeping React at 16.11

### PR DESCRIPTION
**Pull request by mariusandra**
## Changes

- Upgrades PostHog to Kea version `3.0.0-beta.2`. 
- Removes eslint rule for `'no-shadow': 'error',` - it would be impossible to write the new syntax with it
- Removes babel-plugin-kea, since we use typegen (3.0.0) to write paths to logic
- Removes `<Provider>` from everywhere it can
- Replaces the inlined subscriptions plugin with `kea-subscriptions`
- Explicitly connects a lot of logic that were not connected. Kea 3.0 doesn't YOLO-mount when it sees a new logic in a listener.
- Upgrade the logics that used `subscriptions` or `forms` to the kea 3.0 builders syntax.
- Automatically mount `teamLogic` and `organizationLogic` in most jest tests.
- Connect all `prompt` logics.
- Upgrade kea-forms to a version that replaces `validator` with `errors`, update all logics to match
- Had to add `enableFormOnSubmit` to all forms to preserve triggering the submit action on the form's submit (this was the default with last version). 
- Few other tweaks too small to mention

## How did you test this code?

Jest, Cypress, Clicking around everywhere.

### CODE REVIEW

**Review by benjackwhite — APPROVED** [review]
Took a fairly close look through to help me understand the changes in 3.0. Everything looks correct to me and runs locally with a bit of blind moving around the app.
👍

File: frontend/src/scenes/project-homepage/projectHomepageLogic.tsx [hunk 1]
```diff
@@ -1,30 +1,32 @@
...
+    selectors({
         primaryDashboardId: [() => [teamLogic.selectors.currentTeam], (currentTeam) => currentTeam?.primary_dashboard],
         dashboardLogic: [
             (s) => [s.primaryDashboardId],
             (primaryDashboardId): ReturnType<typeof dashboardLogic.build> | null =>
-                dashboardLogic.build(
-                    { id: primaryDashboardId ?? undefined, placement: DashboardPlacement.ProjectHomepage },
-                    false
-                ),
+                dashboardLogic({
```
Comment (Twixes) [inline frontend/src/scenes/project-homepage/projectHomepageLogic.tsx:22]: What's the difference between straight-up initialization and `.build()`?

File: frontend/src/scenes/feature-flags/featureFlagLogic.ts [hunk 2]
```diff
@@ -49,14 +52,14 @@ export interface FeatureFlagLogicProps {
...
 }
 
-export const featureFlagLogic = kea<featureFlagLogicType<FeatureFlagLogicProps>>({
-    path: (key) => ['scenes', 'feature-flags', 'featureFlagLogic', key],
-    props: {} as FeatureFlagLogicProps,
-    key: ({ id }) => id ?? 'new',
-    connect: {
+export const featureFlagLogic = kea<featureFlagLogicType<FeatureFlagLogicProps>>([
+    path(['scenes', 'feature-flags', 'featureFlagLogic']),
+    props({} as FeatureFlagLogicProps),
```
Comment (Twixes) [inline frontend/src/scenes/feature-flags/featureFlagLogic.ts:57]: Would be pretty cool ergonomics-wise to be able to pass the type of props as `props<FeatureFlagLogicProps>()` now (maybe? I'm not sure if that'd actually work with the architecture of those new functions)

**Review by Twixes — APPROVED** [review]
Seems to work as advertised!

File: frontend/src/scenes/project-homepage/projectHomepageLogic.tsx (hunk 1 above)
Comment (mariusandra) [reply frontend/src/scenes/project-homepage/projectHomepageLogic.tsx:22]: It's literally the same thing. https://github.com/keajs/kea/blob/kea-3.0/src/kea/kea.ts#L78-L83

**Pull request merged**